sherpa/csrc/rnnt_beam_search.cc

```diff
@@ -423,4 +417,85 @@ std::vector<std::vector<int32_t>> ModifiedBeamSearch(
   return ans;
 }
 
+k2::FsaVec ShortestPath(k2::FsaVec &lattice) {  // NOLINT
+  k2::Ragged<int32_t> state_batches = k2::GetStateBatches(lattice, true);
+  k2::Array1<int32_t> dest_states = k2::GetDestStates(lattice, true);
+  k2::Ragged<int32_t> incoming_arcs = k2::GetIncomingArcs(lattice, dest_states);
+  k2::Ragged<int32_t> entering_arc_batches =
+      k2::GetEnteringArcIndexBatches(lattice, incoming_arcs, state_batches);
+
+  bool log_semiring = false;
+  k2::Array1<int32_t> entering_arcs;
+  k2::GetForwardScores<float>(lattice, state_batches, entering_arc_batches,
+                              log_semiring, &entering_arcs);
+
+  k2::Ragged<int32_t> best_path_arc_indexes =
+      k2::ShortestPath(lattice, entering_arcs);
+
+  k2::FsaVec out = k2::FsaVecFromArcIndexes(lattice, best_path_arc_indexes);
+  return out;
+}
+
+std::vector<std::vector<int32_t>> FastBeamSearch(
```

Comment:
If I am not mistaken, this function corresponds to : ```fast_beam_search_one_best``` : https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/pruned_transducer_stateless2/beam_search.py#L30
For clarity, it might be better to keep the same function as in icefall, to avoid any misunderstanding, what do you think ?

Reply:
Sorry, we changed our mind, and decided not to write any C++ code for fast_beam_search in sherpa. The idea is we will release the gil in k2's pybind code. Will update this PR soon.  Actually, this won't affect too much of your code already written. Thanks!

Reply:
Yes, we will implement sherpa `fast_beam_search_one_best` in pure python, almost the same as it in icefall.